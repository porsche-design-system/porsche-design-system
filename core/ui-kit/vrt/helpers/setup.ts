import 'jasmine';
import { VisualRegressionTester, VisualRegressionTestOptions } from '@myporsche/myservices-visual-regression-tester';
import * as puppeteer from 'puppeteer';
import { Browser } from 'puppeteer';

let browser: Browser;
let visualRegressionTester: VisualRegressionTester;

const testOptions: VisualRegressionTestOptions = {
  viewports: [320, 480, 760, 1000, 1300, 1760],
  fixturesDir: 'vrt/fixtures',
  resultsDir: 'vrt/results',
  tolerance: 0,
  baseUrl: 'http://localhost:3000/patterns'
};

jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

afterAll(async () => {
  if (browser) {
    await browser.close();
  }
});

export async function getVisualRegressionTester(): Promise<VisualRegressionTester> {
  if (!visualRegressionTester) {
    browser = await puppeteer.launch({
      args: ['--no-sandbox', '--disable-setuid-sandbox']
    });
    visualRegressionTester = new VisualRegressionTester(browser, testOptions);
  }

  return visualRegressionTester;
}
