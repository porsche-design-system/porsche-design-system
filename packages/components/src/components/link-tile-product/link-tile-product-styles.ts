import { buildResponsiveStyles, getCss, type Theme } from '../../utils';
import {
  addImportantToEachRule,
  colorSchemeStyles,
  getFocusJssStyle,
  getHiddenTextJssStyle,
  getThemedColors,
  getTransition,
  hostHiddenStyles,
  hoverMediaQuery,
  prefersColorSchemeDarkMediaQuery,
  preventFoucOfNestedElementsStyles,
} from '../../styles';
import {
  borderRadiusLarge,
  borderRadiusMedium,
  fontHyphenationStyle,
  getMediaQueryMin,
  headingSmallStyle,
  spacingFluidMedium,
  spacingFluidSmall,
  spacingFluidXSmall,
  textXSmallStyle,
  textXXSmallStyle,
} from '@porsche-design-system/styles';
import { type BreakpointCustomizable } from '../../utils/breakpoint-customizable';
import { anchorSlot, headerSlot, type LinkTileProductAspectRatio } from './link-tile-product-utils';
import { type JssStyle } from 'jss';

const slottedAnchorSelector = `a[slot='${anchorSlot}']`;

const anchorJssStyle: JssStyle = {
  position: 'absolute',
  inset: 0,
  zIndex: 1, // necessary to be on top of img
  borderRadius: borderRadiusMedium,
};

const getMultilineEllipsis = (lineClamp: number): JssStyle => {
  return {
    display: '-webkit-box',
    WebkitLineClamp: lineClamp,
    WebkitBoxOrient: 'vertical',
    overflow: 'hidden',
  };
};

export const getComponentCss = (
  hasLikeButton: boolean,
  hasSlottedAnchor: boolean,
  hasPriceOriginal: boolean,
  hasDescription: boolean,
  aspectRatio: BreakpointCustomizable<LinkTileProductAspectRatio>,
  theme: Theme
): string => {
  const { primaryColor, contrastHighColor, contrastMediumColor, backgroundSurfaceColor } = getThemedColors(theme);
  const {
    primaryColor: primaryColorDark,
    contrastHighColor: contrastHighColorDark,
    contrastMediumColor: contrastMediumColorDark,
    backgroundSurfaceColor: backgroundSurfaceColorDark,
  } = getThemedColors('dark');

  return getCss({
    '@global': {
      ':host': {
        display: 'block',
        position: 'relative', // needed for ::slotted(a) to overlay correctly
        ...addImportantToEachRule({
          ...colorSchemeStyles,
          ...hostHiddenStyles,
        }),
      },
      ...preventFoucOfNestedElementsStyles,
      [`slot[name="${headerSlot}"]`]: {
        display: 'block', // to ensure correct like button positioning when slot is unused
      },
      ...addImportantToEachRule({
        '::slotted': {
          ...(hasSlottedAnchor && {
            [`&(${slottedAnchorSelector})`]: {
              ...anchorJssStyle,
              textIndent: '-999999px', // hide anchor label visually but still usable for a11y (only works in RTL-mode because of `overflow: hidden;` parent)
            },
            ...getFocusJssStyle(theme, { slotted: slottedAnchorSelector }),
          }),
          [`&([slot="${headerSlot}"])`]: {
            display: 'flex',
            flexWrap: 'wrap',
            gap: spacingFluidXSmall,
          },
          '&(img), &(picture)': {
            display: 'block',
            width: '100%',
            height: '100%',
            objectFit: 'cover',
            borderRadius: borderRadiusLarge,
            overflow: 'hidden', // needed for picture > img to have correct border-radius
          },
        },
      }),
      ...(hasPriceOriginal && {
        s: {
          color: contrastMediumColor,
          ...prefersColorSchemeDarkMediaQuery(theme, {
            color: contrastMediumColorDark,
          }),
        },
      }),
    },
    root: {
      display: 'flex',
      flexDirection: 'column',
      aspectRatio: '3/4',
      overflow: 'hidden', // TODO: discussable if we should prevent text to overflow .root, â€“ e.g. it also prevents a popover from being shown correctly
      boxSizing: 'border-box',
      borderRadius: borderRadiusMedium,
      padding: spacingFluidSmall,
      color: primaryColor,
      backgroundColor: backgroundSurfaceColor,
      ...buildResponsiveStyles(aspectRatio, (ratio: LinkTileProductAspectRatio) => ({
        aspectRatio: ratio.replace(':', '/'),
      })),
      ...prefersColorSchemeDarkMediaQuery(theme, {
        color: primaryColorDark,
        backgroundColor: backgroundSurfaceColorDark,
      }),
    },
    ...(!hasSlottedAnchor && {
      anchor: {
        ...anchorJssStyle,
        ...getFocusJssStyle(theme),
      },
    }),
    header: {
      display: 'flex',
      gap: spacingFluidSmall,
      justifyContent: 'space-between',
      alignItems: 'flex-start',
    },
    ...(hasLikeButton && {
      button: {
        position: 'relative',
        zIndex: 2, // Necessary to be on top of anchor link
      },
    }),
    image: {
      aspectRatio: '8/9',
      margin: `${spacingFluidSmall} auto ${spacingFluidXSmall}`,
      overflow: 'hidden',
      transition: getTransition('transform', 'moderate'),
      [getMediaQueryMin('s')]: {
        padding: `0 ${spacingFluidMedium}`, // ensures image is not getting to large
      },
      ...hoverMediaQuery({
        '.root:hover &': {
          transform: 'scale3d(1.05,1.05,1.05)',
        },
      }),
    },
    wrapper: {
      display: 'flex',
      flexDirection: 'column',
      margin: 'auto',
      textAlign: 'center',
    },
    heading: {
      margin: '0 0 2px', // ua-style reset
      ...headingSmallStyle,
      ...fontHyphenationStyle,
      ...getMultilineEllipsis(3),
    },
    price: {
      margin: 0, // ua-style reset
      ...textXSmallStyle,
      ...(hasPriceOriginal && {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'center',
        columnGap: spacingFluidXSmall,
      }),
    },
    ...(hasDescription && {
      description: {
        margin: 0, // ua-style reset
        ...textXXSmallStyle,
        ...getMultilineEllipsis(2),
        color: contrastHighColor,
        ...prefersColorSchemeDarkMediaQuery(theme, {
          color: contrastHighColorDark,
        }),
      },
    }),
    ...(hasPriceOriginal && {
      'sr-only': getHiddenTextJssStyle(),
    }),
  });
};
