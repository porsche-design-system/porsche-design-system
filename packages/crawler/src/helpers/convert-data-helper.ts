import type {
  AggregatedData,
  AggregatedTagNamesForVersionsAndPrefixes,
  ConsumedTagNamesForVersionsAndPrefixes,
  TagNameData,
  TagNamesAggregated,
  TagNamesWithPropertyNames,
} from '../types';
import { getUnusedTagNames, incrementTagName } from './count-data-helper';
import { INTERNAL_TAG_NAMES, TAG_NAMES, TagName } from '@porsche-design-system/shared';
import { getComponentMeta } from '@porsche-design-system/component-meta';

export const getPdsTagNamesWithPropertyNames = (): TagNamesWithPropertyNames =>
  Object.values(TAG_NAMES).reduce((result, tagName) => {
    const componentMeta = getComponentMeta(tagName);
    return !INTERNAL_TAG_NAMES.includes(tagName)
      ? {
          ...result,
          [tagName]: componentMeta.props ? Object.keys(componentMeta.props) : [],
        }
      : result;
  }, {} as TagNamesWithPropertyNames);

export const getConsumedPrefixesForVersions = (
  consumedTagNamesForVersions: ConsumedTagNamesForVersionsAndPrefixes
): { [pdsVersion: string]: string[] } =>
  Object.entries(consumedTagNamesForVersions).reduce(
    (result, [pdsVersion, prefixesWithData]) => ({
      ...result,
      [pdsVersion]: Object.keys(prefixesWithData),
    }),
    {}
  );

export const getAggregatedTagNamesWithProperties = (tagNamesWithProperties: TagNameData[]): TagNamesAggregated =>
  tagNamesWithProperties.reduce((result, tagNameData) => {
    const tagName = Object.keys(tagNameData)[0] as TagName;
    result[tagName] = incrementTagName(result[tagName], tagNameData);
    return result;
  }, {} as TagNamesAggregated);

export const getAggregatedConsumedTagNamesForVersionsAndPrefixes = (
  consumedTagNamesForVersions: ConsumedTagNamesForVersionsAndPrefixes
): AggregatedTagNamesForVersionsAndPrefixes =>
  Object.entries(consumedTagNamesForVersions).reduce(
    (result, [pdsVersion, prefixesWithData]) => ({
      ...result,
      [pdsVersion]: Object.entries(prefixesWithData).reduce(
        (result, [prefix, tagNamesWithProperties]) => ({
          ...result,
          [prefix]: getAggregatedConsumedTagNames(tagNamesWithProperties),
        }),
        {}
      ),
    }),
    {}
  );

export const getAggregatedConsumedTagNames = (rawDataWithoutVersionsAndPrefixes: TagNameData[]): AggregatedData => {
  const tagNamesWithPropertiesAggregated = getAggregatedTagNamesWithProperties(rawDataWithoutVersionsAndPrefixes);
  const unusedTagNames = getUnusedTagNames(tagNamesWithPropertiesAggregated);
  return {
    tagNames: tagNamesWithPropertiesAggregated,
    unusedTagNames,
  };
};

export const getRawDataWithoutVersionsAndPrefixes = (
  consumedTagNamesForVersions: ConsumedTagNamesForVersionsAndPrefixes
): TagNameData[] =>
  Object.entries(consumedTagNamesForVersions).reduce(
    (result, [pdsVersion, prefixesWithData]) =>
      result.concat(
        Object.entries(prefixesWithData).reduce(
          (result, [prefix, tagNamesWithProperties]) => result.concat(tagNamesWithProperties),
          [] as TagNameData[]
        )
      ),
    [] as TagNameData[]
  );
