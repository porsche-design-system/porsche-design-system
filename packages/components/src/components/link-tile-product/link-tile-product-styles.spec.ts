import { getComponentCss } from './link-tile-product-styles';
import { validateCssAndMatchSnapshot } from '../../../tests/unit/helpers';

describe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    [true, true, false, false, '3:4', 'light'],
    [true, true, true, false, '3:4', 'light'],
    [true, true, true, true, '3:4', 'light'],
    [true, false, false, false, '3:4', 'dark'],
    [true, false, false, false, '3:4', 'auto'],
    [false, false, false, false, '9:16', 'dark'],
    [true, false, false, false, { base: '3:4', xs: '9:16' }, 'light'],
  ])(
    'should return correct css for hasLikeButton: %s, hasSlottedAnchor: %s, hasPriceOriginal: %s, hasDescription: %s, aspectRatio: %j and theme: %s',
    (...args) => {
      validateCssAndMatchSnapshot(getComponentCss(...args));
    }
  );
});
