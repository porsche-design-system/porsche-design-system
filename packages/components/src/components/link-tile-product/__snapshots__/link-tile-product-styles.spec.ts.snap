// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for hasLikeButton: false, hasSlottedAnchor: false, hasPriceOriginal: false, hasDescription: false, aspectRatio: "9:16" and theme: dark 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 9/16;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #FBFCFF;
  background-color: #212225;
}
.anchor {
  position: absolute;
  inset: 0;
  z-index: 1;
  border-radius: 8px;
}
.anchor::-moz-focus-inner {
  border: 0;
}
.anchor:focus {
  outline: 0;
}
.anchor:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: 2px;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
}
@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;

exports[`getComponentCss() should return correct css for hasLikeButton: true, hasSlottedAnchor: false, hasPriceOriginal: false, hasDescription: false, aspectRatio: "3:4" and theme: auto 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 3/4;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #010205;
  background-color: #EEEFF2;
}
.anchor {
  position: absolute;
  inset: 0;
  z-index: 1;
  border-radius: 8px;
}
.anchor::-moz-focus-inner {
  border: 0;
}
.anchor:focus {
  outline: 0;
}
.anchor:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: 2px;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.button {
  position: relative;
  z-index: 2;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
}
@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media (prefers-color-scheme: dark) {
  .root {
    color: #FBFCFF;
    background-color: #212225;
  }
  .anchor:focus-visible {
    outline-color: #1A44EA;
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;

exports[`getComponentCss() should return correct css for hasLikeButton: true, hasSlottedAnchor: false, hasPriceOriginal: false, hasDescription: false, aspectRatio: "3:4" and theme: dark 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 3/4;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #FBFCFF;
  background-color: #212225;
}
.anchor {
  position: absolute;
  inset: 0;
  z-index: 1;
  border-radius: 8px;
}
.anchor::-moz-focus-inner {
  border: 0;
}
.anchor:focus {
  outline: 0;
}
.anchor:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: 2px;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.button {
  position: relative;
  z-index: 2;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
}
@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;

exports[`getComponentCss() should return correct css for hasLikeButton: true, hasSlottedAnchor: false, hasPriceOriginal: false, hasDescription: false, aspectRatio: {"base":"3:4","xs":"9:16"} and theme: light 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 3/4;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #010205;
  background-color: #EEEFF2;
}
.anchor {
  position: absolute;
  inset: 0;
  z-index: 1;
  border-radius: 8px;
}
.anchor::-moz-focus-inner {
  border: 0;
}
.anchor:focus {
  outline: 0;
}
.anchor:focus-visible {
  outline: 2px solid #1A44EA;
  outline-offset: 2px;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.button {
  position: relative;
  z-index: 2;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
}
@media(min-width:480px) {
  .root {
    aspect-ratio: 9/16;
  }
}

@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;

exports[`getComponentCss() should return correct css for hasLikeButton: true, hasSlottedAnchor: true, hasPriceOriginal: false, hasDescription: false, aspectRatio: "3:4" and theme: light 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted(a[slot='anchor']) {
  position: absolute !important;
  inset: 0 !important;
  z-index: 1 !important;
  border-radius: 8px !important;
  text-indent: -999999px !important;
}
::slotted(a[slot='anchor']::-moz-focus-inner) {
  border: 0 !important;
}
::slotted(a[slot='anchor']:focus) {
  outline: 0 !important;
}
::slotted(a[slot='anchor']:focus-visible) {
  outline: 2px solid #1A44EA !important;
  outline-offset: 2px !important;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 3/4;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #010205;
  background-color: #EEEFF2;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.button {
  position: relative;
  z-index: 2;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
}
@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;

exports[`getComponentCss() should return correct css for hasLikeButton: true, hasSlottedAnchor: true, hasPriceOriginal: true, hasDescription: false, aspectRatio: "3:4" and theme: light 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted(a[slot='anchor']) {
  position: absolute !important;
  inset: 0 !important;
  z-index: 1 !important;
  border-radius: 8px !important;
  text-indent: -999999px !important;
}
::slotted(a[slot='anchor']::-moz-focus-inner) {
  border: 0 !important;
}
::slotted(a[slot='anchor']:focus) {
  outline: 0 !important;
}
::slotted(a[slot='anchor']:focus-visible) {
  outline: 2px solid #1A44EA !important;
  outline-offset: 2px !important;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
s {
  color: #6B6D70;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 3/4;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #010205;
  background-color: #EEEFF2;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.button {
  position: relative;
  z-index: 2;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  column-gap: clamp(4px, 0.25vw + 3px, 8px);
}
.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0,0,0,0);
  white-space: nowrap;
}
@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;

exports[`getComponentCss() should return correct css for hasLikeButton: true, hasSlottedAnchor: true, hasPriceOriginal: true, hasDescription: true, aspectRatio: "3:4" and theme: light 1`] = `
":host {
  display: block;
  position: relative;
  color-scheme: light dark !important;
}
:host([hidden]) {
  display: none !important;
}
slot[name="header"] {
  display: block;
}
::slotted(a[slot='anchor']) {
  position: absolute !important;
  inset: 0 !important;
  z-index: 1 !important;
  border-radius: 8px !important;
  text-indent: -999999px !important;
}
::slotted(a[slot='anchor']::-moz-focus-inner) {
  border: 0 !important;
}
::slotted(a[slot='anchor']:focus) {
  outline: 0 !important;
}
::slotted(a[slot='anchor']:focus-visible) {
  outline: 2px solid #1A44EA !important;
  outline-offset: 2px !important;
}
::slotted([slot="header"]) {
  display: flex !important;
  flex-wrap: wrap !important;
  gap: clamp(4px, 0.25vw + 3px, 8px) !important;
}
::slotted(img), ::slotted(picture) {
  display: block !important;
  width: 100% !important;
  height: 100% !important;
  object-fit: cover !important;
  border-radius: 12px !important;
  overflow: hidden !important;
}
s {
  color: #6B6D70;
}
.root {
  display: flex;
  flex-direction: column;
  aspect-ratio: 3/4;
  overflow: hidden;
  box-sizing: border-box;
  border-radius: 8px;
  padding: clamp(8px, 0.5vw + 6px, 16px);
  color: #010205;
  background-color: #EEEFF2;
}
.header {
  display: flex;
  gap: clamp(8px, 0.5vw + 6px, 16px);
  justify-content: space-between;
  align-items: flex-start;
}
.button {
  position: relative;
  z-index: 2;
}
.image {
  aspect-ratio: 8/9;
  margin: clamp(8px, 0.5vw + 6px, 16px) auto clamp(4px, 0.25vw + 3px, 8px);
  overflow: hidden;
  transition: transform var(--p-transition-duration, 0.4s) cubic-bezier(0.25,0.1,0.25,1);
}
.wrapper {
  display: flex;
  flex-direction: column;
  margin: auto;
  text-align: center;
}
.heading {
  margin: 0 0 2px;
  font: normal normal 600 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}
.price {
  margin: 0;
  font: normal normal 400 clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  column-gap: clamp(4px, 0.25vw + 3px, 8px);
}
.description {
  margin: 0;
  font: normal normal 400 .75rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  color: #535457;
}
.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0,0,0,0);
  white-space: nowrap;
}
@media(min-width:760px) {
  .image {
    padding: 0 clamp(16px, 1.25vw + 12px, 36px);
  }
}

@media(hover:hover) {
  .root:hover .image {
    transform: scale3d(1.05,1.05,1.05);
  }
}
"
`;
